### PR TITLE
docs: design the documentation directory structure

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2bgyq1g SDK 文档基础架构搭建](https://app.clickup.com/t/2bgyq1g)

File: docs/source/index.rst [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+
+   features/dataset_management/index
+   features/version_control/index
+   features/sheet_management/index
+
+.. toctree::
+   :maxdepth: 1
+   :caption: ADVANCED FEATURES
+
+   advanced_features/storage_config
```
Comment (linjiX) [inline docs/source/index.rst:23]: shall we hide `Advance Features` since there are no content under it now?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: docs/source/index.rst (hunk 1 above)
Comment (graczhual) [reply docs/source/index.rst:23]: ok

**Review by Lee-000 — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

File: docs/source/index.rst [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+   getting_started/installation
+   getting_started/10_minutes_to_graviti
+
+.. toctree::
+   :maxdepth: 1
+   :caption: FEATURES
+
+   features/dataset_management/index
+   features/version_control/index
+   features/sheet_management/index
```
Comment (AChenQ) [inline docs/source/index.rst:17]: Shall we move the `sheet_mamagement` under the `dataset_management`? 
It might look more ordered.

**Review by AChenQ — APPROVED** [review]

**Pull request merged**